### ISSUE TITLE
Broken dependencies

### ISSUE THREAD

**Issue by tomwimmenhove** [tensorflow/models#10334]
When trying to install the research model on Ubuntu 20.04 (and I'm guessing other's. Since it's using pip, I'm guessing every install is affected) you'll run into dependency issues as shown below:

## Using Docker:
```
$ docker build -f research/object_detection/dockerfiles/tf2/Dockerfile -t od .
...
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorflow-gpu 2.2.0 requires gast==0.3.3, but you have gast 0.4.0 which is incompatible.
tensorflow-gpu 2.2.0 requires h5py<2.11.0,>=2.10.0, but you have h5py 3.1.0 which is incompatible.
tensorflow-gpu 2.2.0 requires tensorboard<2.3.0,>=2.2.0, but you have tensorboard 2.7.0 which is incompatible.
tensorflow-gpu 2.2.0 requires tensorflow-estimator<2.3.0,>=2.2.0, but you have tensorflow-estimator 2.6.0 which is incompatible.
```

## Using pip:
```
~/Tensorflow/models/research$ python -m pip install --use-feature=2020-resolver .

Usage:   
  /usr/bin/python -m pip install [options] <requirement specifier> [package-index-options] ...
  /usr/bin/python -m pip install [options] -r <requirements file> [package-index-options] ...
  /usr/bin/python -m pip install [options] [-e] <vcs project url> ...
  /usr/bin/python -m pip install [options] [-e] <local project path> ...
  /usr/bin/python -m pip install [options] <archive url/path> ...

no such option: --use-feature
```

I don't know if this is a pip version issue. Running it without ```--use-feature``` yields:

```
~/Tensorflow/models/research$ python -m pip install .
...
ERROR: apache-beam 2.33.0 has requirement avro-python3!=1.9.2,<1.10.0,>=1.8.1, but you'll have avro-python3 1.10.2 which is incompatible.
ERROR: tensorflow-metadata 1.2.0 has requirement absl-py<0.13,>=0.9, but you'll have absl-py 0.15.0 which is incompatible.
ERROR: google-auth-httplib2 0.1.0 has requirement httplib2>=0.15.0, but you'll have httplib2 0.14.0 which is incompatible.
ERROR: google-api-python-client 2.27.0 has requirement httplib2<1dev,>=0.15.0, but you'll have httplib2 0.14.0 which is incompatible.
```

**Comment by mouhannadali** [comment]
I have the same issue and made it work with "pip install --upgrade pip" but model training loss is worse than if I run the same code in Colab. any idea?

**Comment by jvishnuvardhan** [comment]
@tomwimmenhove @mouhannadali Is this still an issue with recent TF versions. Can you please check with recent versions and let us know whether the issue persists with recent versions? Thanks
